query.go

```diff
@@ -255,6 +255,15 @@ func (q *Query) SetBuffer(attributeOrDimension string, buffer interface{}) (*uin
 					err)
 			}
 
+			hasDim, err := domain.HasDimension(attributeOrDimension)
```

Comment:
This should be above, the idea is to not call `schema.AttributeFromName` if its a dimension. We should check if its a dim or a attr before trying to fetch the object. This apply everywhere.

Reply:
According to TileDB export comments: `tiledb_domain_has_dimension` : `Checks whether the domain has a dimension of the given name.` Will it work with attribute names?

Reply:
There is also `tiledb_array_schema_has_attribute`.